### DISCUSSION TITLE
在使用基于服务端身份认证的Token拦截器的时候如何在发起请求前确定Token过期状态

### DISCUSSION THREAD

**Discussion by viilon** [alovajs/alova#387] [Q&A]
在某些特定场景下发起请求前就已经知道token肯定是过期状态了，例如本地存储的token为空时，在这种场景下就不需要在等待响应后来判断token是否为过期状态了。assignToken选项可以在请求前做一些操作，但是无法设置token的状态。这种场景下该如何解决呀，求大佬解答

**Comment by JOU-amjs** [comment]
这个问题有两种方案
第一种，也是最常见的做法，就是在路由钩子中判断如果没有登录则直接重定向到登录页，不会触发请求。

第二种，不常见做法，通过请求前钩子判断是否登录，未登录则跳转登录页并抛出错误阻止继续请求。
